libs/seiscomp/processing/waveformprocessor.cpp

```diff
@@ -284,6 +284,15 @@ bool WaveformProcessor::store(const Record *record) {
 	if ( _stream.lastRecord ) {
 		if ( record == _stream.lastRecord ) return false;
 
+		const auto minGapThres = 2 * 1.0 / _stream.fsamp;
```

Comment:
As this check only depends on _stream.fsamp it would be more efficient to only check for it when _stream.fsamp is being set. That is done in `initFilter`. Would you agree?

Reply:
That's true. Though, then the naming of the `initFilter` method is kind of misleading. So I would do it before this block here https://github.com/SeisComP/common/blob/0932f584e44c600d84cfded3ecf4f0f8affbf7b9/libs/seiscomp/processing/waveformprocessor.cpp#L315-L322

Reply:
On the other hand, what happens if the sampling frequency changes?:
* filters should be reinitialized
* the gap threshold should be checked once again

Reply:
Yes, `initFilter` should actually be called `initStream` ;) You can also do it before this block. Changing sampling rates are not handled currently. Once this is implemented, a new function `setupStream` should be called.